packages/next/server/dev/hot-reloader.ts

```diff
@@ -435,15 +439,15 @@ export default class HotReloader {
     startSpan.stop() // Stop immediately to create an artificial parent span
 
     await this.clean(startSpan)
+
     // Ensure distDir exists before writing package.json
-    await fs.mkdir(this.config.distDir, { recursive: true })
+    await fs.mkdir(this.distDir, { recursive: true })
```

Comment:
here `fs.mkdir` was trying to create a file from the root folder to a relative path that starts with `../../` which goes outside of the project.
therefore, I've replaced with `distDir` from the place of the Instance creation, where `distDir` is an absolute path and the issue dissappeared.